proto/proto/block_header.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+package block_header;
+
+import "digest.proto";
+
+message BlockHeader {
+    /// the hash of the previous blocks header.
+    digest.Digest prev_hash = 1;
+    /// a unique sequential number of the current block.
+    uint64 block_num = 2;
```

Comment:
A `u32` is probably enough for a block number.

Reply:
changed